examples/platform/esp32/time/TimeSync.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <esp_sntp.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+class Esp32TimeSync
+{
+public:
+    Esp32TimeSync(char * aSntpServerName, uint16_t aSyncSntpIntervalDay)
+    {
+        mSntpServerName = new char[strlen(aSntpServerName) + 1];
```

Comment:
as a best practice, please also do cleanup of this in a destructor. Or you could use a `ScopedMemoryString` as a member.

Reply:
When I use destructor or ScopedMemoryString, my copied string gets deleted and I failed to get the real time form ntp.

Reply:
Made `mSntpServerName` as global variable.